### PR TITLE
split tests by install type

### PR DESCRIPTION
**Pull request by briangallagher**
# Description
Refactor e2e and functional tests based on install types. 2 categories, all install types for rhmi 2.X and the managed-api install type.

## Verify
**Install managed-api**

1. Install the managed-api operator and verify
1.1 export INSTALLATION_TYPE=managed-api
2. Run the functional tests make test/functional
3. Verify all the tests in the managed api install type test suite ran

Also running here: 
https://master-jenkins-csb-intly.cloud.paas.psi.redhat.com/job/Playground/job/briang-test-managed-apis/

**Install rhmi**

1. Install the rhmi operator and verify
1.1 export INSTALLATION_TYPE=rhmi
2. Run the functional tests make test/functional
3. Verify all the tests in the rhmi install type test suite ran

Also running here: 
https://master-jenkins-csb-intly.cloud.paas.psi.redhat.com/job/Playground/job/briang/

### CODE REVIEW

**Comment by psturc** [comment]
👀

File: test/functional/aws_shared_functions.go [hunk 1]
```diff
@@ -26,39 +27,45 @@ const (
 	awsCredsSecretName = "aws-creds"
 )
 
-var (
+func getExpectedPostgres(installationName string) []string {
```
Comment (sergioifg94) [inline test/functional/aws_shared_functions.go:30]: Should we have some logic in this function to check the install type and return a different list depending on it?

**Review by sergioifg94 — COMMENTED** [review]
Code looks good. Just a few things

**Comment by psturc** [comment]
@briangallagher when running this against `managed-api` cluster, I saw some tests were failing. I lost the full list of failed tests (I can run it again and post it here later), but I remember `C04` was failing because of the missing alert rules (that are valid only for RHMI):
```
github.com/integr8ly/integreatly-operator  MGDAPI-46-tests-by-install-type ✔                                                                                                                                                             6d
▶ go clean -testcache && go test -v ./test/functional -run="//^C04 - Verify Alerts exist" -timeout=80m
...
=== RUN   TestIntegreatly/Integreatly_Happy_Path_Tests/C04_-_Verify_Alerts_exist
time="2020-09-22T14:25:08+02:00" level=info msg="Looking for Rhmi CR in redhat-rhmi-operator namespace"
    TestIntegreatly/Integreatly_Happy_Path_Tests/C04_-_Verify_Alerts_exist: alerts_exist.go:802:
        File Name: redhat-rhmi-amq-online-ksm-endpoint-alerts.yaml
    TestIntegreatly/Integreatly_Happy_Path_Tests/C04_-_Verify_Alerts_exist: alerts_exist.go:803: Missing Rules: [RHMIAMQOnlineNoneAuthServiceEndpointDown RHMIAMQOnlineAddressSpaceControllerServiceEndpointDown RHMIAMQOnlineConsoleServiceEndpointDown RHMIAMQOnlineRegistryCsServiceEndpointDown RHMIAMQOnlineStandardAuthServiceEndpointDown RHMIAMQOnlineEnmasseOperatorMetricsServiceEndpointDown]
    TestIntegreatly/Integreatly_Happy_Path_Tests/C04_-_Verify_Alerts_exist: alerts_exist.go:804: Unexpected Rules: []
    TestIntegreatly/Integreatly_Happy_Path_Tests/C04_-_Verify_Alerts_exist: alerts_exist.go:805: Status: File expected but not found
    TestIntegreatly/Integreatly_Happy_Path_Tests/C04_-_Verify_Alerts_exist: alerts_exist.go:802:
        File Name: redhat-rhmi-codeready-workspaces-ksm-endpoint-alerts.yaml
...
```

also there were failing tests from the `FUNCTIONAL_TESTS` suite.

**Comment by sergioifg94** [comment]
I'm seeing the same failure as @psturc. I think this is because the `expectedRules` list in the `alerts_exist.go` test haven't been split

File: test/functional/aws_shared_functions.go (hunk 1 above)
Comment (trepel) [reply test/functional/aws_shared_functions.go:30]: I can confirm that this is an issue for Managed API nightly pipeline. The test fails like this:
```--- FAIL: TestIntegreatly/Integreatly_Happy_Path_Tests/F01_-_Verify_AWS_rds_resources_exist_and_are_in_expected_state (2.85s)
            aws_rds_resources_exist.go:20: test cro postgres exists failed with the following errors : [
                failed to find codeready-postgres-rhmi postgres cr : postgres.integreatly.org "codeready-postgres-rhmi" not found 
                found codeready-postgres-rhmi postgres not ready with phase: , message:  
                codeready-postgres-rhmi postgres cr does not contain a resource id annotation: annotations for  can not be nil 
                failed to find ups-postgres-rhmi postgres cr : postgres.integreatly.org "ups-postgres-rhmi" not found 
                found ups-postgres-rhmi postgres not ready with phase: , message:  
                ups-postgres-rhmi postgres cr does not contain a resource id annotation: annotations for  can not be nil 
                failed to find fuse-postgres-rhmi postgres cr : postgres.integreatly.org "fuse-postgres-rhmi" not found 
                found fuse-postgres-rhmi postgres not ready with phase: , message:  
                fuse-postgres-rhmi postgres cr does not contain a resource id annotation: annotations for  can not be nil]```

Comment (trepel) [reply test/functional/aws_shared_functions.go:30]: This failure is due to very same reason:
```--- FAIL: TestIntegreatly/Integreatly_Happy_Path_Tests/A21_-_Verify_AWS_maintenance_and_backup_windows (307.94s)```

Comment (briangallagher) [reply test/functional/aws_shared_functions.go]: Code updated to address this

File: test/common/operand_versions.go [hunk 2]
```diff
@@ -30,16 +30,37 @@ var (
...
+		integreatlyv1alpha1.CloudResourcesStage: {
+			integreatlyv1alpha1.ProductCloudResources: integreatlyv1alpha1.VersionCloudResources,
+		},
+		integreatlyv1alpha1.ProductsStage: {
+			integreatlyv1alpha1.Product3Scale:    integreatlyv1alpha1.Version3Scale,
+			integreatlyv1alpha1.ProductRHSSOUser: integreatlyv1alpha1.VersionRHSSOUser,
+		},
+		integreatlyv1alpha1.SolutionExplorerStage: {
+			integreatlyv1alpha1.ProductSolutionExplorer: integreatlyv1alpha1.VersionSolutionExplorer,
+		},
```
[the remark this answers is not in the document]
Comment (briangallagher) [reply test/common/operand_versions.go]: Code updated to address this

File: test/common/operator_versions.go [hunk 3]
```diff
@@ -29,18 +29,38 @@ var (
...
+			integreatlyv1alpha1.ProductCloudResources: integreatlyv1alpha1.OperatorVersionCloudResources,
+		},
+		integreatlyv1alpha1.ProductsStage: {
+			integreatlyv1alpha1.Product3Scale:    integreatlyv1alpha1.OperatorVersion3Scale,
+			integreatlyv1alpha1.ProductRHSSOUser: integreatlyv1alpha1.OperatorVersionRHSSOUser,
+		},
+
+		integreatlyv1alpha1.SolutionExplorerStage: {
+			integreatlyv1alpha1.ProductSolutionExplorer: integreatlyv1alpha1.OperatorVersionSolutionExplorer,
+		},
```
[the remark this answers is not in the document]
Comment (briangallagher) [reply test/common/operator_versions.go]: Code updated to address this

**Comment by briangallagher** [comment]
@sergioifg94 @psturc Thanks for review
With regard to the test failures

I've updated C04_-_Verify_Alerts_exist and the AWS postgres issue above. I think i messed up a merge. 
I believe Verify_3scale_CRUDL_permissions failed due to C04_-_Verify_Alerts_exist failing.

Running locally a number of tests will fail,
- All AWS tests
- SMTP related tests

Given the above exceptions, that should be all tests accounted for.

**Comment by sergioifg94** [comment]
Finished running the tests after the changes and other than the expected failures I only got a failure in _`A16_-_Custom_first_broker_login_flow`_. This test passed before so I'd say it's flaky

/lgtm

**Comment by philbrookes** [comment]
should we be overriding the e2e test for this PR then?

**Comment by matskiv** [comment]
ci/prow/versions test log mentions some "CONFLICT"
so I would do a retest to be on the safe side..

/retest

**Pull request merged**